src/twisted/trial/runner.py

```diff
@@ -258,26 +271,31 @@ def run(self, result):
             self._bail()
 
 
-def name(thing):
+_Loadable: TypeAlias = Union[
+    modules.PythonAttribute,
+    modules.PythonModule,
+    pyunit.TestCase,
+    Type[pyunit.TestCase],
+]
+
+
+def name(thing: _Loadable) -> str:
     """
     @param thing: an object from modules (instance of PythonModule,
         PythonAttribute), a TestCase subclass, or an instance of a TestCase.
     """
-    if isTestCase(thing):
+    if isinstance(thing, pyunit.TestCase):
+        return thing.id()
+    elif isinstance(thing, (modules.PythonAttribute, modules.PythonModule)):
+        return thing.name
+    elif isTestCase(thing):
```

Comment:
very minor style comment ... while reading the `assert False` 


```suggestion
    if isinstance(thing, pyunit.TestCase):
        return thing.id()

    if isinstance(thing, (modules.PythonAttribute, modules.PythonModule)):
        return thing.name

    if isTestCase(thing):
```

Reply:
The `assert False` in the `else` of this construction makes mypy happy that all branches are covered :/  I don't know why it can't figure it out that the if-elif cases cover everything.  I think `isTestCase(...)` may be tripping it up but I don't understand why.